app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,83 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
+  # GET /posts
+  # GET /posts.json
+  def index
+    if user_signed_in?
+      current_user.user_status = current_user.posts.count > 2 ? 'active' : 'idle'
+      current_user.save
+    end
+    @posts = Post.all.order('created_at DESC')
+    @post = Post.new
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show; end
+
+  # GET /posts/new
+  def new
+    @post = current_user.posts.build
+  end
```

Comment:
:bulb: In the `new` method, the conventional way to initialize a post is as follows:
`@post = Post.new`
Since you are already using `build` in the create method, we don't need it here.